Integrations/MicrosoftGraphListener/MicrosoftGraphLIstener.py

```diff
@@ -0,0 +1,982 @@
+        :rtype: ``list``
+        """
+
+        attachment_results = []  # type: ignore
+        suffix_endpoint = f'users/{self._mailbox_to_fetch}/messages/{message_id}/attachments'
+        attachments = self._http_request('Get', suffix_endpoint).get('value', [])
+
+        for attachment in attachments:
+            attachment_type = attachment.get('@odata.type', '')
+            attachment_name = attachment.get('name', 'untitled_attachment')
+            if attachment_type == self.FILE_ATTACHMENT:
+                try:
+                    attachment_content = base64.b64decode(attachment.get('contentBytes', ''))
+                except Exception as e:  # skip the uploading file step
+                    demisto.info(f"MS-Graph-Listener: failed in decoding base64 file attachment with error {str(e)}")
+                    continue
+            elif attachment_type == self.ITEM_ATTACHMENT:
+                attachment_id = attachment.get('id', '')
+                attachment_content = self._get_attachment_mime(attachment_id)
+                attachment_name = f'{attachment_name}.eml'
+            # upload the item/file attachent to War Room
+            upload_file(attachment_name, attachment_content, attachment_results)
+
+        return attachment_results
+
+    def _parse_email_as_incident(self, email):
+        """
+        Parses fetched emails as incidents.
+
+        :type email: ``dict``
+        :param email: Fetched email to parse
+
+        :return: Parsed email
+        :rtype: ``dict``
+        """
+        parsed_email = MsGraphClient._parse_item_as_dict(email)
+
+        if email.get('hasAttachments', False):  # handling attachments of fetched email
+            parsed_email['Attachments'] = self._get_email_attachments(message_id=email.get('id', ''))
+
+        incident = {
+            'name': parsed_email['Subject'],
+            'details': email.get('bodyPreview', '') or parsed_email['Body'],
+            'labels': MsGraphClient._parse_email_as_labels(parsed_email),
+            'occurred': parsed_email['ReceivedTime'],
+            'attachment': parsed_email.get('Attachments', []),
+            'rawJSON': json.dumps(parsed_email)
+        }
+
+        return incident
+
+    @logger
+    def fetch_incidents(self, last_run):
+        """
+        Fetches emails from office 365 mailbox and creates incidents of parsed emails.
+
+        :type last_run: ``dict``
+        :param last_run:
+            Previous fetch run data that holds the fetch time in utc Y-m-dTH:M:SZ format,
+            ids of fetched emails, id and path of folder to fetch incidents from
+
+        :return: Next run data and parsed fetched incidents
+        :rtype: ``dict`` and ``list``
+        """
+        start_time = get_now_utc()
+        last_fetch = last_run.get('LAST_RUN_TIME', None)
+        exclude_ids = last_run.get('LAST_RUN_IDS', [])
+        last_run_folder_path = last_run.get('LAST_RUN_FOLDER_PATH')
+        folder_path_changed = False
```

Comment:
this way you don't need to update it later
```suggestion
        folder_path_changed = last_run_folder_path != self._folder_to_fetch
```